src/main/webapp/app/overview/exercise-details/course-exercise-details.component.ts

```diff
@@ -118,9 +124,64 @@ export class CourseExerciseDetailsComponent implements OnInit, OnDestroy {
         this.exercise.participationStatus = participationStatus(this.exercise);
         this.isAfterAssessmentDueDate = !this.exercise.assessmentDueDate || moment().isAfter(this.exercise.assessmentDueDate);
         this.exerciseCategories = this.exerciseService.convertExerciseCategoriesFromServer(this.exercise);
+        if (this.exercise.type === ExerciseType.PROGRAMMING) {
+            this.getProgrammingExercise();
+        }
         this.subscribeForNewResults();
     }
 
+    /**
+     * The button should be only visible in the dev environment
+     */
+    get isInProduction(): boolean {
+        return this.profileService.isInProduction();
+    }
+
+    /**
+     * checks if exercise has no local setup
+     */
+    get hasNoLocalSetup(): boolean {
+        return this.programmingExerciseSimulationUtils.hasNoLocalSetup(this.programmingExercise.testRepositoryUrl);
+    }
+
+    /**
+     * asks the server for the programming exercise with the provided exercise ID
+     */
+    getProgrammingExercise() {
+        this.courseExerciseSubmissionResultSimulationService.getProgrammingExercise(this.exerciseId).subscribe((programmingExercise) => {
+            this.programmingExercise = programmingExercise;
+        });
+    }
+
+    /**
+     * triggers the simulation of a participation and submission for the currently logged in user
+     */
+    simulateSubmission() {
+        this.courseExerciseSubmissionResultSimulationService.simulateSubmission(this.exerciseId).subscribe(
```

Comment:
add guard for production misuse
if (isInProduction) {
console.log("error");
this.jhiAlertService.error('This action is NOT supported on production and should NOT be visible. Please contact a developer immediately!')
}

Reply:
I added guards for the methods, which should be not used in production